app/assets/javascripts/analytics/ecommerce.js

```diff
@@ -0,0 +1,66 @@
+(function () {
+  "use strict";
+  window.GOVUK = window.GOVUK || {};
+
+  var Ecommerce = function (config) {
+    this.init = function (element) {
+      this.forEachEcommerceRow(element, this._sendImpression);
+      this.forEachEcommerceRow(element, this._trackProductClick);
+    }
+
+    this.forEachEcommerceRow = function (element, fct) {
+      var ecommerceRowCss = '[data-ecommerce-row]';
+      var ecommerceRows = element.find(ecommerceRowCss);
+      var startPosition = parseInt(element.data('ecommerce-start-index'));
+
+      ecommerceRows.each(function(index, ecommerceRow) {
+        var $ecommerceRow = $(ecommerceRow);
+
+        var content_id = $ecommerceRow.attr('data-ecommerce-content-id'),
+          path = $ecommerceRow.attr('data-ecommerce-path');
+
+        fct($ecommerceRow, content_id, path, index + startPosition);
+      });
+    }
+
+    this._sendImpression = function (_row, content_id, path, position) {
+      // We only send the id to GA as additional product data is linked when it is uploaded.
+      // This approach is taken to avoid the GA data packet exceeding the 8k limit
+      ga('ec:addImpression', {
+        id: content_id || path,
+        position: position,
+        list: 'SITE_SEARCH_RESULTS'
+      });
+    }
+
+    this._trackProductClick = function (row, content_id, path, position) {
+      row.click(function(event) {
+        ga('ec:addProduct', {
+          id: content_id || path,
+          position: position
+        });
+
+        ga('ec:setAction', 'click', {list: 'SITE_SEARCH_RESULTS'});
+        ga('send', 'event', 'UX', 'click', 'Results',
+          GOVUK.CustomDimensions.getAndExtendDefaultTrackingOptions({}));
+      })
+    }
+  }
+
+  Ecommerce.ec_loaded = false;
+  Ecommerce.start = function (element) {
+    element = element || $('[data-analytics-ecommerce]');
+
+    if(element.length > 0) {
+      if(!Ecommerce.ec_loaded) {
+        ga('require', 'ec');
+        Ecommerce.ec_loaded = true;
+      }
+      var ecommerce = new Ecommerce();
+      ecommerce.init(element);
+    }
+  }
+
+  window.GOVUK.Ecommerce = Ecommerce;
+  window.GOVUK.StaticAnalytics.beforeTrackPage = Ecommerce.start;
```

Comment:
Instead of doing this, can't we just directly invoke `Ecommerce.start` from `static-analytics.js`? Then we wouldn't need to introduce this slightly obscure `beforeTrackPage` mechanism, and the dependency is explicit at the desired point of invocation.